apps/site/scripts/orama-search/get-documents.mjs

```diff
@@ -1,80 +1,78 @@
-import { existsSync, readFileSync } from 'node:fs';
-import { join } from 'node:path';
-import zlib from 'node:zlib';
-
-import { slug } from 'github-slugger';
-
-import { getRelativePath } from '../../next.helpers.mjs';
-
-const currentRoot = getRelativePath(import.meta.url);
-const dataBasePath = join(currentRoot, '../../.next/server/app/en/next-data');
-
-if (!existsSync(dataBasePath)) {
-  throw new Error(
-    'The data directory does not exist. Please run `npm run build` first.'
+import { readFile, glob } from 'node:fs/promises';
+import { join, basename } from 'node:path';
+
+import generateReleaseData from '#site/next-data/generators/releaseData.mjs';
+import { getRelativePath } from '#site/next.helpers.mjs';
+
+import { processDocument } from './process-documents.mjs';
+
+// If a GitHub token is available, include it for higher rate limits
+const fetchOptions = process.env.GITHUB_TOKEN
+  ? { headers: { Authorization: `Bearer ${process.env.GITHUB_TOKEN}` } }
+  : undefined;
+
+/**
+ * Fetch Node.js API documentation directly from GitHub
+ * for the current Active LTS version.
+ */
+const getAPIDocs = async () => {
+  // Find the current Active LTS version
+  const releaseData = await generateReleaseData();
+  const version = releaseData.find(
+    r => r.status === 'Active LTS'
+  ).versionWithPrefix;
+
+  // Get list of API docs from the Node.js repo
+  const fetchResponse = await fetch(
+    `https://api.github.com/repos/nodejs/node/contents/doc/api?ref=${version}`,
+    fetchOptions
+  );
+  const documents = await fetchResponse.json();
+
+  // Download and return content + metadata for each doc
+  return Promise.all(
+    documents.map(async ({ name, download_url }) => {
+      const res = await fetch(download_url, fetchOptions);
+
+      return processDocument({
+        content: await res.text(),
+        pathname: `docs/${version}/api/${basename(name, '.md')}.html`,
+      });
+    })
   );
-}
-
-const nextPageData = readFileSync(`${dataBasePath}/page-data.body`, 'utf-8');
```

Comment:
The structuring of the response content seems quite different from the old version to the new one. Is this intentional?